mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/AbstractCheckerCmd.scala

```diff
@@ -5,46 +5,55 @@ import org.backuity.clist.{arg, opt}
 import java.io.File
 import com.typesafe.scalalogging.LazyLogging
 import at.forsyte.apalache.infra.passes.options.SourceOption
+import at.forsyte.apalache.infra.passes.options.Config
+import at.forsyte.apalache.io.ConfigurationError
 
 // Holds the minimal necessary info about a specification.
 abstract class AbstractCheckerCmd(val name: String, description: String)
     extends PassExecutorCmd(name, description) with LazyLogging {
   var file: File = arg[File](description = "a file containing a TLA+ specification (.tla or .json)")
-  var config: String = opt[String](name = "config", default = "", description = "configuration file in TLC format")
-  var cinit: String = opt[String](name = "cinit", default = "",
+  var config: Option[String] =
+    opt[Option[String]](name = "config", default = None, description = "configuration file in TLC format")
+  var cinit: Option[String] = opt[Option[String]](name = "cinit", default = None,
       description = "the name of an operator that initializes CONSTANTS,\n" +
         "default: None")
-  var init: String = opt[String](name = "init", default = "",
+  var init: Option[String] = opt[Option[String]](name = "init", default = None,
       description = "the name of an operator that initializes VARIABLES,\n" +
         "default: Init")
-  var next: String =
-    opt[String](name = "next", default = "", description = "the name of a transition operator, default: Next")
-  var inv: String =
-    opt[String](name = "inv", default = "", description = "the name of an invariant operator, e.g., Inv")
-  var temporal: String =
-    opt[String](name = "temporal", default = "", description = "the name of a temporal property, e.g. Property")
-  var length: Int =
-    opt[Int](name = "length", default = 10, description = "maximal number of Next steps, default: 10")
+  var next: Option[String] =
+    opt[Option[String]](name = "next", default = None, description = "the name of a transition operator, default: Next")
+  var inv: Option[String] =
+    opt[Option[String]](name = "inv", default = None, description = "the name of an invariant operator, e.g., Inv")
+  var temporal: Option[String] =
+    opt[Option[String]](name = "temporal", default = None,
+        description = "the name of a temporal property, e.g. Property")
+  var length: Option[Int] =
+    opt[Option[Int]](name = "length", default = None, description = "maximal number of Next steps, default: 10")
 
+  override def toConfig(): Config.ApalacheConfig = {
+    val cfg = super.toConfig()
+    cfg.copy(
+        common = cfg.common.copy(file = Some(file)),
+        checker = cfg.checker.copy(config = config, cinit = cinit, init = init, next = next, inv = inv,
+            temporal = temporal, length = length),
+    )
+  }
   override def setCommonOptions(): Unit = {
+    // TODO: rm once OptionProvider is wired in
+    val cfg = configuration.left.map(err => new ConfigurationError(err)).toTry.get
```

Comment:
This is an intermediate step, using the derived configuration to set all values in the option map. In a follow up, we will remove the option map and replace it with a statically verifiable class of options.